### PR TITLE
#IMQiZKj6 Logout from account

### PR DESCRIPTION
**Pull request by karigirwa**
### **What about this PR?**
This PR is about to fix setup Logout account

### **Description of Task to be completed**
- Authenticated user should be able to logout.
- Authenticated user should be able to access protected resources/URLs
- Appropriate error messages are displayed on unsuccessful attempts

### **How should be manually tested?**
- Open the swagger documentation through this [link](https://phantom-caba-ft-logout--az7r73.herokuapp.com/api-docs/)
- Login as admin with credentials below or register and login with your credentials
       {
              email: admin@phantom.com
              password: admin
      }
- Use provided token to logout

### **Relevant pivots tracker**

- [IMQiZKj6](https://trello.com/c/S9EuF0ch/30-logout-from-account)

### CODE REVIEW

File: src/test/jwt.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  it('should return JSON web token', () => {
+    const token = signinToken(claims);
+    mockToken= token
+    chai.expect(token).equal(token);
+  });
+  it('should return object including claims', () => {
+    const payload = decode(mockToken);
+    chai.expect(payload).equal(payload);
+  });
+});
```
Comment (Nkbtemmy) [inline src/test/jwt.test.js]: remember to add new line

File: src/Middleware/Auth.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+//         if (!roles.includes(req.user.role)) {
+//             return next(
+//                 res.status(403).json({
+//                     message: res.__('permission for this action')
+//                 })
+//             )
+//         }
+//         next()
+//     }
+// }
```
Comment (Nkbtemmy) [inline src/Middleware/Auth.js]: new line required

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]

**Comment by ceelogre** [comment]
To be revisited in the frontend.

**Comment by Nkbtemmy** [comment]
how can i test this task functionality, while you don't write its documentation ????

**Review by Niyiragira — DISMISSED** [review]

File: src/controller/userControllers.js [hunk 3]
```diff
@@ -221,6 +227,14 @@ class userController {
       return res.status(500).send(error.message);
     }
   }
+  static async logout(req,res){
+    try {
+      localStorage.removeItem("token");
```
Comment (ceelogre) [inline src/controller/userControllers.js]: localstorage should remain a browser implementation. Remove it from backend.

File: src/middleware/protectRoutes.js [hunk 4]
```diff
@@ -1,12 +1,16 @@
 import jwt from 'jsonwebtoken'
 import Model from '../database/models'
+import localStorage from 'localStorage'
 
 exports.protect = async (req, res, next) => {
 
-    let token 
+    let token = localStorage.getItem('token')
```
Comment (ceelogre) [inline src/middleware/protectRoutes.js]: See previous review comment.

**Review by ceelogre — CHANGES_REQUESTED** [review]

File: src/middleware/protectRoutes.js [hunk 5]
```diff
@@ -1,12 +1,16 @@
 import jwt from 'jsonwebtoken'
 import Model from '../database/models'
+import localStorage from 'localStorage'
```
Comment (ceelogre) [inline src/middleware/protectRoutes.js]: What are you using localstorage for?

**Review by ceelogre — CHANGES_REQUESTED** [review]

**Review by ceelogre — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]

**Comment by Niyiragira** [comment]
After logout, when I use the token to get data from the protected routes, I still get the data. I should  not.

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]
This PR was good but you can still access protected routers after you logout

**Review by Niyiragira — APPROVED** [review]

**Review by Nkbtemmy — APPROVED** [review]

**Review by blaise82 — APPROVED** [review]

**Pull request merged**